stablehlo/dialect/StablehloOps.td

```diff
@@ -643,12 +643,12 @@ def StableHLO_TanhOp: StableHLO_UnaryElementwiseOp<"tanh",
 //===----------------------------------------------------------------------===//
 // See https://www.tensorflow.org/xla/operation_semantics#element-wise_binary_arithmetic_operations
 
-class StableHLO_BinaryElementwiseOp<string mnemonic, list<Trait> traits> :
-    StableHLO_Op<mnemonic, traits # [InferShapedTypeOpInterface,
-    SameOperandsAndResultShape, Elementwise]> {
+class StableHLO_BinaryElementwiseOp<string mnemonic, list<Trait> traits,
+    Type OperandType = HLO_Tensor, Type ResultType = OperandType> :
+    StableHLO_Op<mnemonic, traits # [InferShapedTypeOpInterface, SameOperandsAndResultShape]> {
```

Comment:
Is there a particular reason to push `[Elementwise]` down to subclasses?

Reply:
All other ops should have this Trait, but I've lowered it to individual ops since ComplexOp has different operand/result element type:
https://github.com/llvm/llvm-project/blob/main/mlir/lib/IR/Operation.cpp#L1085-L1090
```
stablehlo.complex' op all non-scalar operands/results must have the same shape and base type
```

Reply:
Nvm. Not explicitly providing OperandType and ResultType (HLO_Fp32Or64Tensor and HLO_ComplexTensor) removes the need to lower the `[Elementwise]` trait down to the subclasses.